spark/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -131,6 +143,21 @@
     this.splitLookback = options.get("lookback").map(Integer::parseInt).orElse(null);
     this.splitOpenFileCost = options.get("file-open-cost").map(Long::parseLong).orElse(null);
 
+    if (io.getValue() instanceof HadoopFileIO) {
+      String scheme = "no_exist";
+      try {
+        FileSystem fs = new Path(table.location()).getFileSystem(
+            SparkSession.active().sparkContext().hadoopConfiguration());
```

Comment:
Do we need to get a FileSystem instance to get the scheme of the table location? Seems like we might be able to just parse it as a URI, or use `split(":")[0]` instead. That would avoid getting the active Spark context.

Reply:
This is minor and can be done in a follow-up.